### PR TITLE
Some CI progess

### PR DESCRIPTION
**Pull request by fjebaker**
See commit message for details.

### CODE REVIEW

**Comment by fjebaker** [comment]
Not ready to merge, I have progress, am seeing how far I can get.

**Comment by fjebaker** [comment]
Okay here's what I've worked out:
- Disabling TLS I can see in wireshark that the test suite works fine, all the communications are going through, and voila the test suite does indeed work.
- Enabling TLS, wireshark can't decrypt TLS v1.3 even with RSA keys, but I can see the packets going backwards and forwards, and I get a `CERTIFICATE_VERIFY_FAILED` on the new test I wrote.
- Using `MDB_DISABLE_HOSTNAME_VERIFY`, I again observe packets go backwards and forwards, but now the client crashes trying to do the `readuntil(seperator=END_BYTES)` check, which means it burns through all its connection attempts and then implodes.

**Comment by fjebaker** [comment]
Out of time right now but recommend debugging with
```bash
pytest -vvv -k test_ping_launcher
```
to just trigger one test. Also I've been running these in a docker container too, to get closer to the actions environment. Not necessarily useful, but also kind of a little bit useful.

**Comment by fjebaker** [comment]
If you're happy with these changes @TomMelt, merge and let the CI spin on your branch and let's iterate from there. I get all tests running in my docker container now, though the output `assert` is failing due to some differences.

File: tests/utils.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    def __enter__(self) -> "BackgroundProcess":
+        print("--- LAUNCHING ---")
+        self._running_command = Popen(
+            self.command,
+            stdout=PIPE,
+            stderr=PIPE,
+            env={
+                # on this CI this might run as root because
+                # docker containers are the wild wild west
```
Comment (TomMelt) [inline tests/utils.py:21]: :joy:

**Review by TomMelt — APPROVED** [review]
Thanks @fjebaker I will give it a whirl

**Pull request merged**